cmd/gitserver/server/serverutil.go

```diff
@@ -616,26 +616,26 @@ func isPaused(dir string) (string, bool) {
 	return string(b), true
 }
 
-// bestEffortWalk is a filepath.Walk which ignores errors that can be passed
+// bestEffortWalk is a filepath.WalkDir which ignores errors that can be passed
 // to walkFn. This is a common pattern used in gitserver for best effort work.
 //
 // Note: We still respect errors returned by walkFn.
 //
 // filepath.Walk can return errors if we run into permission errors or a file
 // disappears between readdir and the stat of the file. In either case this
 // error can be ignored for best effort code.
-func bestEffortWalk(root string, walkFn func(path string, info fs.FileInfo) error) error {
-	logger := log.Scoped("bestEffortWalk", "bestEffortWalk is a filepath.Walk which ignores errors that can be passed to walkFn")
-	return filepath.Walk(root, func(path string, info fs.FileInfo, err error) error {
+func bestEffortWalk(root string, walkFn func(path string, entry fs.DirEntry) error) error {
+	if msg, ok := isPaused(root); ok {
+		logger := log.Scoped("bestEffortWalk", "bestEffortWalk is a filepath.Walk which ignores errors that can be passed to walkFn")
```

Comment:
to make it similar to the function doc?
```suggestion
		logger := log.Scoped("bestEffortWalk", "bestEffortWalk is a filepath.WalkDir which ignores errors that can be passed to walkFn")
```